editor/plugins/path_3d_editor_plugin.cpp

```diff
@@ -37,6 +37,13 @@
 #include "node_3d_editor_plugin.h"
 #include "scene/resources/curve.h"
 
+static bool _is_in_handle(int p_id, int p_num_points) {
+	int t = (p_id + 1) % 2;
+	int idx = (p_id + 1) / 2;
+	// order of points is [out_0, out_1, in_1, out_2, in_2, ... out_n-1, in_n-1, in_n]
+	return (t == 1) != (idx == 0 || idx == (p_num_points - 1));
```

Comment:
It would be nice to rewrite this complex expression into something more readable, probably spliting it into `if (idx == 0) { ... } else if (idx == (p_num_points - 1) { ... } else { ... }`